src/address.c

```diff
@@ -68,15 +68,15 @@ int coap_is_mcast(const coap_address_t *a) {
   }
  return 0;
 }
-#else /* !defined(WITH_CONTIKI) && !defined(WITH_LWIP) */
 
-#ifdef __clang__
-/* Make compilers happy that do not like empty modules. As this function is
- * never used, we ignore -Wunused-function at the end of compiling this file
- */
-#pragma GCC diagnostic ignored "-Wunused-function"
+#endif /* !defined(WITH_CONTIKI) && !defined(WITH_LWIP) */
+
+void coap_address_init(coap_address_t *addr) {
+  assert(addr);
+  memset(addr, 0, sizeof(coap_address_t));
+#if !defined(WITH_LWIP) && !defined(WITH_CONTIKI)
+  /* lwip and Contiki have constant address sizes and doesn't need the .size part */
```

Comment:
Let's take the opportunity and fix the grammar: "doesn't" → "don't".

Reply:
Updated and pushed